lib/moog-require.js

```diff
@@ -62,7 +63,12 @@ module.exports = function(options) {
         self._indexes = glob.sync(self.options.localModules + '/**/index.js');
       }
       const matches = self._indexes.filter(function(index) {
-        return index.endsWith('/' + type + '/index.js');
+        // Double-check that we're not confusing "@apostrophecms/asset" with "asset" by
+        // making sure "type" is not preceded by an npm namespace folder. If type is itself namespaced
+        // this never comes up (because npm namespaces don't nest). The risk is that a legitimate
+        // project level module that happens to end with the same name as a namespaced module
+        // will be rejected as a duplicate when nestedModuleSubdirs is present
+        return index.endsWith('/' + type + '/index.js') && !index.match(new RegExp(`/@[^/]+/${regExpQuote(type)}/index\\.js$`));
```

Comment:
If taking https://github.com/apostrophecms/apostrophe/pull/3736/files#r863610696 into account:

```suggestion
        return index.endsWith(`/${type}/index.js`);
```